components/builder-api/src/server/resources/projects.rs

```diff
@@ -270,37 +269,46 @@ fn update_project((req, body): (HttpRequest<AppState>, Json<ProjectUpdateReq>))
         Err(err) => return err.into(),
     };
 
-    let mut project_get = OriginProjectGet::new();
-    project_get.set_name(format!("{}/{}", &origin, &name));
-
-    let mut project = match route_message::<OriginProjectGet, OriginProject>(&req, &project_get) {
-        Ok(project) => project,
-        Err(err) => return err.into(),
-    };
-
-    let mut request = OriginProjectUpdate::new();
-
     if body.plan_path.len() <= 0 {
         return HttpResponse::with_body(
             StatusCode::UNPROCESSABLE_ENTITY,
             "Missing value for field: `plan_path`",
         );
     }
 
+    let conn = match req.state().db.get_conn() {
+        Ok(conn_ref) => conn_ref,
+        Err(_) => return HttpResponse::new(StatusCode::INTERNAL_SERVER_ERROR),
+    };
+
+    // TODO (SA): We should not need to fetch the origin project here.
```

Comment:
This TODO is why the migrations file is exploding. So many unnecessary calls to get ID's